### ISSUE TITLE
Converting a old Gallery block to individual Image blocks does not work

### ISSUE THREAD

**Issue by mrfoxtalbot** [Automattic/wp-calypso#59496]
### Quick summary

If you have a **Gallery block** that was created with versions of Gunteberg older than 12.1.0 and try to convert it to individual Image blocks, it will not work.

Instead of creating multiple Image blocks, one per image in the gallery, it will replace the entire gallery with a single empty block, as seen below:

![Kapture 2021-12-22 at 14 16 45](https://user-images.githubusercontent.com/4452464/147100405-ba92fd5a-4db8-4115-94a5-9e7874cb7335.gif)

You can use this sample code of a pre-Version 12.1.0 gallery to recreate this error on any simple site:

```
<!-- wp:gallery {"linkTo":"none"} -->
<figure class="wp-block-gallery columns-3 is-cropped"><ul class="blocks-gallery-grid"><li class="blocks-gallery-item"><figure><img src="https://mrfoxtestingsite.files.wordpress.com/2020/11/laik.jpg?w=788" alt="" data-id="3715" class="wp-image-3715"/></figure></li><li class="blocks-gallery-item"><figure><img src="https://mrfoxtestingsite.files.wordpress.com/2020/11/laik-4.jpg?w=1024" alt="" data-id="3714" class="wp-image-3714"/></figure></li><li class="blocks-gallery-item"><figure><img src="https://mrfoxtestingsite.files.wordpress.com/2020/11/laik-3.jpg?w=1024" alt="" data-id="3713" class="wp-image-3713"/></figure></li><li class="blocks-gallery-item"><figure><img src="https://mrfoxtestingsite.files.wordpress.com/2020/11/laik-2.jpg?w=1024" alt="" data-id="3712" class="wp-image-3712"/></figure></li></ul></figure>
<!-- /wp:gallery -->
```

This is **only happening on Simple sites**. I was not able to reproduce this on an AT site or on a self hosted site (with or without the Gutenberg plugin Version 12.1.0)

### Steps to reproduce

1. Visit a post where you have a gallery. This post must have been crated using an older version of Gutenberg (before 12.1.0). If you do not have any galleries, you can use the code above.
2. Try to convert that gallery into image blocks

### What you expected to happen

It should create one image block per image in the gallery, with the correct image assigned.

### What actually happened

The Gallery was replaced by an empty image block.

### Context

I noticed this while looking at some old draft posts.

### Simple, Atomic or both?

Simple

### Theme-specific issue?

N/A

### Browser, operating system and other notes

Mac OS Big Sur + Chrome 96.0.4664.110

### Reproducibility

Consistent

### Severity

Most (> 50%)

### Workaround details

No workaround, really.

**Comment by mrfoxtalbot** [comment]
Related to https://github.com/WordPress/gutenberg/issues/36818 maybe?

**Comment by Robertght** [comment]
This was discussed briefly here: p1640179670059400-slack-C01UW7SB411 however, I wasn't able to replicate this.

PInging @glendaviesnz for a second look. Does this appear to be familiar with the new gallery/old gallery compatibility?

**Comment by fullofcaffeine** [comment]
I'm not able to replicate, either 🤔 . Here's a screencast of my attempt to reproduce:

https://user-images.githubusercontent.com/81248/147131624-079ee9e6-7ac1-4713-9d1c-3466f95a57e9.mp4

I've tested with 12.2 (upcoming release) and 12.1.

@mrfoxtalbot Are you able to reproduce this consistently? Do you see any errors in the JS console? Also wondering if @glendaviesnz might have any insights about this?

**Comment by mrfoxtalbot** [comment]
Hi @fullofcaffeine! Yes, I can reproduce it consistently on simple sites but please note that **it is only happening if the gallery block was created with an older version of Gutenberg**.

You can test it yourself on this post:
https://mrfoxtestingsite.wordpress.com/wp-admin/post.php?post=3558&action=edit
and on this post:
https://rebelkfest.wordpress.com/wp-admin/post.php?post=2495&action=edit
I have invited you to both sites, in case you do not have superadmin access. They are my own testing sites, feel free to play around.

I see this error in the console. but I am not 100% it is relevant:
`DevTools failed to load source map: Could not load content for chrome-extension://liecbddmkiiihnedobmlmillhodjkdmb/js/intercom-link-expand-loader.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME`
![Screen Shot on 2021-12-22 at 18:39:09](https://user-images.githubusercontent.com/4452464/147134018-f5237a83-b52a-4123-8aa4-4d6b0bb2afec.png)

**Comment by fullofcaffeine** [comment]
Thanks for the additional details, Alvaro!

> Yes, I can reproduce it consistently on simple sites but please note that it is only happening if the gallery block was created with an older version of Gutenberg.

Do you happen to know what version? I did try creating the gallery in 11.9.1, as you can see in the screencast above.

**Comment by mrfoxtalbot** [comment]
The date on the posts I had tried was September 2020 or ealier. That being said, I seem unable to reproduce it now. It would appear that once I visit the post, even if I do not make any changes to it, the "Old" Gallery Block gets replaced it with a new one. In fact, I cannot reproduce it anymore on ANY of the posts where I noticed it originally 🤔 . If I try adding the sample code I shared in the original post, it gets added as a classic post.

I think we can close this as "Cannot reproduce"... but I suspect certain old posts might still be affected by this.

**Comment by glendaviesnz** [comment]
@mrfoxtalbot looking at your screen recording it looks like for some reason the gallery has not been automatically updated to v2 as there is an `Update` option in the block toolbar:

<img width="695" alt="Screen Shot 2021-12-23 at 10 30 41 AM" src="https://user-images.githubusercontent.com/3629020/147157982-39ec0cd0-a978-49b8-a882-792622687dcd.png">

It looks like there was an issue with transforming the V1 gallery blocks to Image blocks and I was able to replicate this on a local install with an old version of Gutenberg. However in all instances when I tried it on WPcom with 12.2 the Gallery was automatically migrated to the v2 format (even using the sample code that you provide above), and the transform to Image blocks works as expected.

I am not sure what caused your example to not migrate to v2 automatically 🤔. Given that everything works as expected with the v2 format, and by default, everything on WPcom should now be automatically migrating to v2 I don't think it is worth trying to track down the cause of this. If other instances are encountered the workaround is to click the `Update` button in the Gallery block toolbar and then transform should work.
